src/coreclr/tools/r2rdump/CoreDisTools.cs

```diff
@@ -1209,6 +1209,217 @@ private static bool IsAnotherRuntimeFunctionWithinMethod(int rva, RuntimeFunctio
+            {
+                ReplaceRelativeOffset(ref instruction, rtf.StartAddress + rtfOffset + offs, rtf);
+            }
+            else if (IsLoongArch64JirlRAInstruction(instr, out uint rj, out int imm))
+            {
+                // Common Pattern:
+                //      pcaddu12i
+                //      ld.d
+                //      jirl  ra, rj, 0
+                //
+                //      pcaddu12i
+                //      addi.d
+                //      ld.d
+                //      jirl  ra, rj, 0
+                //  There may exist some irrelevant instructions between pcaddu12i and jirl.
+                //  We need to find relevant instructions based on rj to calculate the jump address.
+                uint register  = rj;
+                int  immediate = imm;
+                int  target    = 0;
+                bool isFound   = false;
+                int currentInsOffs = rtfOffset - InstructionSize;
+                int currentPC  = rtf.StartAddress + currentInsOffs;
+
+                do
+                {
+                    instr = BitConverter.ToUInt32(_reader.Image, imageOffset + currentInsOffs);
+
+                    if (IsLoongArch64Ld_dOrAddi_dInstruction(instr, out uint rd, out rj, out imm))
+                    {
+                        if (rd == register)
+                        {
+                            register = rj;
+                            immediate += imm;
+                        }
+                    }
+                    else if (IsLoongArch64Pcaddu12iInstruction(instr, out rd, out imm))
+                    {
+                        if (rd == register)
+                        {
+                            immediate += currentPC + imm;
+                            isFound = true;
+                            break;
+                        }
+                    }
+                    else
+                    {
+                        // check if target register is using by an unexpected instruction.
+                        rd = (instr & 0x1f);
+                        if ((rd == register) && !IsLoongArch64FLD_DInstruction(instr))
+                        {
+                            break;
+                        }
+                    }
+
+                    currentInsOffs -= InstructionSize;
+                    currentPC      -= InstructionSize;
+                } while (currentInsOffs > 0);
+
+                if (isFound)
+                {
+                    if (!TryGetImportCellName(target + immediate, out string targetName) || string.IsNullOrWhiteSpace(targetName))
+                    {
+                        return;
+                    }
+
+                    StringBuilder updatedInstruction = new();
+                    updatedInstruction.Append(instruction);
+                    updatedInstruction.Append(" // ");
+                    updatedInstruction.Append(targetName);
+                    instruction = updatedInstruction.ToString();
```

Comment:
nit: this can be simplified to:
```suggestion
                    instruction = $"{instruction} // {targetName}";
```

Reply:
Thanks very much!

Reply:
cc @SzpejnaDawid this can also applies for RISC-V.